servers/rendering/rendering_server_canvas.cpp

```diff
@@ -113,7 +113,12 @@ void RenderingServerCanvas::_cull_canvas_item(Item *p_canvas_item, const Transfo
 	}
 
 	Rect2 rect = ci->get_rect();
-	Transform2D xform = p_transform * ci->xform;
+	Transform2D xform = ci->xform;
+	if (snapping_2d_transforms_to_pixel) {
+		xform.elements[2].floor();
```

Comment:
I've just been backporting this to 3.2, and noticed, there is an error here which stops the whole method from working.

This line needs to be `xform.elements[2] = xform.elements[2].floor();`, as it stands, this line is a no-op. I've made this mistake more than once with godot funcs myself! :grin: